system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.hpp

```diff
@@ -107,8 +99,34 @@ class PeriodicMeasurementNode : public system_metrics_collector::Collector, publ
    * If true, then measurements are cleared after publishing data
    */
   const bool clear_measurements_on_publish_;
+
+  rclcpp::Time measurement_start_;
   rclcpp::TimerBase::SharedPtr measurement_timer_;
   rclcpp::TimerBase::SharedPtr publish_timer_;
+
+protected:
+  /**
+   * Creates a ROS2 timer with a period of measurement_period_.
+   *
+   * @return if setup was successful
+   */
+  bool setupStart() override;
```

Comment:
`setupStart` should be final here, we should also have a pure virtual method called `onStart()` which is protected here.